src/Control/Distributed/Spark/RDD.hs

```diff
@@ -79,6 +89,35 @@ map clos rdd = do
     f <- reflect clos
     call rdd "map" [coerce f]
 
+mapPartitions
+    :: ( Reflect (Closure (Int32 -> Iterator a -> Iterator b)) ty
```

Comment:
This looks wrong: don't you want `Closure (Iterator a -> Iterator b)` unlike for `mapPartitionWithIndex`? I'm suprised that this type checks.